src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -765,7 +902,10 @@ trait AppAndGroupFormats {
     * made it optional (also with ip-per-container in mind) and we have to re-add it in cases where it makes sense.
     */
   private[this] def addHealthCheckPortIndexIfNecessary(app: AppDefinition): AppDefinition = {
-    val hasPortMappings = app.container.exists(_.docker.exists(_.portMappings.exists(_.nonEmpty)))
+    val hasPortMappings = app.container.exists(_ match {
+      case docker: Container.Docker => docker.pms.nonEmpty
```

Comment:
This should be delegated to the container. E.g. MesosDocker could support this in the future.